src/KGzocha/Searcher/Searcher.php

```diff
@@ -3,62 +3,67 @@
 namespace KGzocha\Searcher;
 
 use KGzocha\Searcher\Context\SearchingContextInterface;
-use KGzocha\Searcher\FilterImposer\Collection\FilterImposerCollectionInterface;
-use KGzocha\Searcher\FilterModel\Collection\FilterModelCollectionInterface;
-use KGzocha\Searcher\FilterModel\FilterModelInterface;
+use KGzocha\Searcher\QueryCriteriaBuilder\Collection\QueryCriteriaBuilderCollectionInterface;
+use KGzocha\Searcher\QueryCriteria\Collection\QueryCriteriaCollectionInterface;
+use KGzocha\Searcher\QueryCriteria\QueryCriteriaInterface;
 
+/**
+ * Class Searcher
+ *
+ * @package KGzocha\Searcher
+ */
 class Searcher implements SearcherInterface
 {
     /**
-     * @var FilterImposerCollectionInterface
+     * @var QueryCriteriaBuilderCollectionInterface
      */
-    private $imposerCollection;
+    private $queryCriteriaBuilderCollection;
 
     /**
      * @var SearchingContextInterface
      */
     private $searchingContext;
 
     /**
-     * @param FilterImposerCollectionInterface $imposerCollection
+     * @param QueryCriteriaBuilderCollectionInterface $queryCriteriaBuilderCollection
      * @param SearchingContextInterface $searchingContext
      */
     public function __construct(
-        FilterImposerCollectionInterface $imposerCollection,
+        QueryCriteriaBuilderCollectionInterface $queryCriteriaBuilderCollection,
         SearchingContextInterface $searchingContext
     ) {
-        $this->imposerCollection = $imposerCollection;
+        $this->queryCriteriaBuilderCollection = $queryCriteriaBuilderCollection;
         $this->searchingContext = $searchingContext;
     }
 
     /**
      * @inheritdoc
      */
     public function search(
-        FilterModelCollectionInterface $filterCollection
+        QueryCriteriaCollectionInterface $filterCollection
     ) {
-        foreach ($filterCollection->getImposedModels() as $filterModel) {
+        foreach ($filterCollection->getCriteria() as $filterModel) {
```

Comment:
Just to remember: we need to change variable names from `$filterModel` to `$criteria`